pkg/isoeditor/stream.go

```diff
@@ -75,6 +75,26 @@ func ignitionOverlay(isoPath string, ignitionContent *IgnitionContent, allowOver
 	if err != nil {
 		return nil, nil, errors.Wrap(err, "failed to create overwrite reader for ignition")
 	}
+
+	if ibf.info.Length > ibf.dataSize {
+		offset, _, err := GetISOFileInfo(ibf.info.File, isoPath)
+		if err != nil {
+			r.Close()
+			return nil, nil, err
+		}
+		paddingLen := ibf.info.Length - ibf.dataSize
+		paddingOverlay := overlay.Overlay{
+			Reader: bytes.NewReader(bytes.Repeat([]byte{0}, int(paddingLen))),
+			Offset: offset + ibf.info.Offset + ibf.dataSize,
```

Comment:
I may be misunderstanding, but this isn't safe to do in a streaming context, right?
Wouldn't we need to expand the size of the entire ISO to ensure that nothing was overwritten after the ignition embed area?

Maybe this will only ever be done if we've already set `allowOverflow`, but that's not clear from this new code.

Reply:
It's always safe to overwrite from the end of the data to the end of the embed area with zeroes. That's what this commit does.

It's only safe to to extend the embed area if allowOverflow is true, but that is handled by the other commit in findBoundaries().